vowpalwabbit/v_array.h

```diff
@@ -83,6 +101,27 @@ struct v_array
     }
   }
 
+  // insert() and remove() don't follow the standard spec, which calls for iterators
+  // instead of indices. But these fn signatures follow our usage better.
+  // These functions do not check bounds, undefined behavior if they are called
+  // on out-of-bounds indices
+  // insert before the indexed element
+  inline void insert(size_t idx, const T& elem)
+  {
+    if (_end == end_array)
+      resize(2 * capacity() + 3);
+    _end++;
+    memmove(&_begin[idx + 1], &_begin[idx], (size() - (idx + 1)) * sizeof(T));
+    _begin[idx] = elem;
+  }
+  // erase indexed element
+  inline void erase(size_t idx)
+  {
+    _begin[idx].~T();
+    memmove(&_begin[idx], &_begin[idx + 1], (size() - (idx + 1)) * sizeof(T));
```

Comment:
Undefined behavior for non trivially copyable types right? Can you add a comment